server/src/internalClusterTest/java/org/elasticsearch/health/HealthMetadataServiceIT.java

```diff
@@ -93,7 +105,7 @@ public void testWatermarkSettingUpdate() throws Exception {
             int numberOfNodes = 3;
             ByteSizeValue randomBytes = ByteSizeValue.ofBytes(randomLongBetween(6, 19));
             String initialWatermark = percentageMode ? randomIntBetween(86, 94) + "%" : randomBytes.toString();
-            ByteSizeValue initialMaxHeadroom = percentageMode ? randomBytes : ByteSizeValue.MINUS_ONE;
+            ByteSizeValue initialMaxHeadroom = randomBytes;
```

Comment:
Is percentage mode not needed here anymore? Below at line 140 we check it when we do the assertion

Reply:
in this line, no, not needed anymore. This setting has logic when getting the default value, depending upon a different setting. While checking this test, the old behavior was incorrect, using `-1` just to make the test pass. The percentage logic at line 107, will either trigger or not the default-getter-logic